nix/docker/default.nix

```diff
@@ -0,0 +1,63 @@
+{ postgrest, dockerTools, writeShellScriptBin }:
+let
+  image =
+    tag:
+      dockerTools.buildImage {
+        inherit tag;
+
+        name = "postgrest/postgrest";
+        contents = postgrest;
+
+        # Set the current time as the image creation date. This makes the build
+        # non-reproducible, but that should not be an issue for us.
+        created = "now";
+
+        extraCommands =
+          ''
+            mkdir etc
+            cp ${./postgrest.conf} etc/postgrest.conf
+          '';
```

Comment:
How about the non-root group/user? I saw you added those in: https://github.com/monacoremo/postgrest/commit/694fd927a8c71fe819852623ec99bd8c1fa5f98e#diff-5712e736e0de6ba170577f8472c398e9R92-R93.

For reference, we have them in our [Dockerfile](https://github.com/PostgREST/postgrest/blob/master/docker/Dockerfile#L47-L48)

Reply:
Yes, that should already be the case :-) It runs as non-root based on the line `User = "1000";`. 

In the bare bones image, actually creating that user in `/etc/passwd`, `shadow` etc. does not seem to be necessary, as there is only our static binary in it and nothing else that could possibly inspect those files.

The only case that I can think of where that could make a difference is when `libpq` tries to infer the database login user name from the local user. This would resolve to `postgrest` in the current image and fail here. I don't think its likely that someone relies on this, however, and even if, it would be easily fixed by explicitly setting the username.

Reply:
Looked into that a bit more, I found a few references that the user does not have to actually exist:
* https://docs.docker.com/engine/reference/run/#user
> When passing a numeric ID, the user does not have to exist in the container.
* [Openshift](https://docs.openshift.com/container-platform/4.4/openshift_images/create-images.html#use-uid_create-images) (more details in the [older docs](https://docs.openshift.com/container-platform/3.3/creating_images/guidelines.html#use-uid) run containers under arbitrary user ids that may or may not exist as users in the container

This should work well this way!

Reply:
I see. I forgot we also specify `USER 1000` in the Dockerfile.

So adding the postgrest user/group through `useradd/groupadd` wouldn't be necessary anymore?

Reply:
Yes, it's not needed in this case as there is only postgrest in the image and any uid has execute permissions on it. Thats exactly enough to run the application correctly!

There is no other program like bash, whoami etc. in the image that would look at /etc/passwd, so we don't actually need it.